src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -63,8 +63,8 @@ bool ClientRequest::is_pg_op() const
 void ClientRequest::may_set_prev_op()
 {
   // set prev_op_id if it's not set yet
-  if (__builtin_expect(prev_op_id == 0, true)) {
-    prev_op_id = sequencer.get_last_issued();
+  if (__builtin_expect(prev_op_id == std::nullopt, true)) {
```

Comment:
```suggestion
  if (__builtin_expect(prev_op_id.has_value(), true)) {
```

Reply:
Using `has_value()` would impose a negation. Also, my impression is that `std::nullopt` is far more specific to `std::optional` than `has_value()`, and thus makes easier to recognize the `optional` is underneath.

Reply:
IMHO, `!prev_op_id.has_value()` is more readable than `prev_op_id == std::nullopt`, also, we can put
```c++
if (__builtin_expect(prev_op_id.has_value(), false)) {
  // ...
}
```

Reply:
i don't think we should emphasis the type underneath of a value, what matter is its meaning in this context.

Reply:
[Changing the second argument of `__builtin_expect` doesn't change the logic](https://godbolt.org/z/6Mf3qPzoP); it only tells compiler which branch of the conditional is expected. With `has_value()` we would need to:
a) negate
```cpp
if (__builtin_expect(!prev_op_id.has_value(), true)) {
  // ...
}
```
b) or early exit.

Reply:
i'd vote for negate.

Reply:
OK!